buzhash.go

```diff
@@ -8,23 +8,32 @@ import (
 )
 
 const (
-	buzMin  = 128 << 10
-	buzMax  = 512 << 10
-	buzMask = 1<<17 - 1
+	buzMinDefault = 16 << 10
+	buzMaxDefault = 64 << 10
+	buzMinLegacy  = 128 << 10
+	buzMaxLegacy  = 512 << 10
+	buzMask       = 1<<17 - 1
```

Comment:
Note that this affects the "target" size, which is the average distance after the min size where chunk boundaries will be found. This setting gives tgt=2^17 or 128K. This means the average chunk boundary will be at 128K + 16K = 144K. Since this is larger than your max of 64K, this means most chunks will be truncated to 64K.

You need to also set `busMaskLegacy=1<<17 -1` and change `buzMaskDefaul=1<<14 -1` for tgt=16K for a default average block size of 32K. I'd also bump up `buzMaxDefault = 128 << 10`

Reply:
Shouldn't that be (1<<14)-1?

Reply:
Ha! It turns out `<<` does have higher precedence than `-`. See https://github.com/ipfs/go-ipfs/pull/8952#discussion_r868722640